crates/matrix-sdk/src/widget/permissions.rs

```diff
@@ -23,4 +23,11 @@ pub struct Permissions {
     pub read: Vec<EventFilter>,
     /// Types of the messages that a widget wants to be able to send.
     pub send: Vec<EventFilter>,
+    /// If this permission is requested by the widget the client should not
+    /// provide the option to pop the widget into its own window.
+    ///
+    /// If the widget does not work without the widget api (e.g. it relies on
+    /// asking for matrix room messages) it will(/has to) request this
+    /// permission.
```

Comment:
How about this then?

```suggestion
    /// If this permission is requested by the widget, it can not operate
    /// separately from the matrix client.
    ///
    /// This means clients should not offer to open the widget in a separate
    /// browser.
```

Reply:
Yes this is good. I changed `browser` to make it more clear what is meant by this but other then that used your proposal:
```
    /// If this permission is requested by the widget, it can not operate
    /// separately from the matrix client.
    ///
    /// This means clients should not offer to open the widget in a separate
    /// browser/tab/webview that is not connected to the postmessage widget-api.
    ```